docs/schemaform/form-config.md

```diff
@@ -0,0 +1,297 @@
+    // to the property name. It will follow our ExpandingGroup pattern and expand underneath
+    // the field it is set to.
+    expandUnder: '', 
+    
+    // Set this if you want to hide this field on the review page.
+    hideOnReview: true || false,
+
+    // Set this if you want to hide this field on the review page when the field value is falsy
+    hideOnReviewIfFalse: true || false 
+
+    // Function that conditionally hides fields in the form
+    hideIf: function (fieldData) {
+      return true || false;
+    }
+
+    // Function that conditionally replaces the current field's schema
+    updateSchema: function (fieldData, pageData) {
+      return {};
+    }
+  }
+}
+```
+
+## Schemaform cookbook
+
+Here are some common situations you might run into when building a form and how to address them.
+
+### I need to write custom validation
+
+JSON Schema does not provide all the validation options we need in our forms, so we've created an additional way to add field validations, using `ui:validations` in the uiSchema object. `ui:validations` is an array and each item can be a function or an object. If you pass a function, it will be called with the following arguments:
+
+- errors: The errors object for the field.
+- fieldData: The data for the field.
+- pageData: The current form (page) data.
+- schema: The current JSON Schema for the field.
+- errorMessages: The error messsage object (if available) for thie field.
+
+Every validation function should update the errors object with any errors found. This is done by calling its `addErrors()` method. Here's an example:
+
+```js
+function validateSSN(errors, ssn) {
+  if (!isValidSSN(ssn)) {
+    errors.addError('Please enter a valid nine digit SSN (dashes allowed)');
+  }
+}
+```
+
+Items in the `ui:validations` array can also be objects. Objects should have two properties:
+
+- options: Object (or anything, really) that will be passed to your validation function. You can use this to allow your validation function to be configurable for different fields on the form.
+- validator: A function with the same signature as above, plus the options object.
+
+### I need to validate a field based on other fields in the same object
+
+You don't have to limit your use of `ui:validations` to non-object fields (i.e. the ones that become visible inputs on the form). You can also validate objects, which allows you to compare subfields. For example, given this schema:
+
+```js
+{ 
+  type: 'object',
+  properties: {
+    email: {
+      type: 'string'
+    },
+    confirmEmail: {
+      type: 'string'
+    }
+  }
+}
+```
+
+If you use `ui:validations` on this object field (instead of on the email or confirmEmail fields) you can compare the two fields:
+
+```js
+export function validateEmailsMatch(errors, formData) {
+  const { email, confirmEmail } = formData;
+  if (email !== confirmEmail) {
+    errors.confirmEmail.addError('Please ensure your entries match');
+  }
+}
+```
```

Comment:
Can you extend this example by showing how you would then use `validateEmailsMatch`?